### PR TITLE
Reduce discovery times to 10 seconds and do not store duplicated events

### PR DESCRIPTION
**Pull request by fabriziosestito**
This PR reduces the discovery time to 5 seconds and prevent the collector to store duplicated events.

If a duplicate is found, we keep sending it to the projectors in case a projection had failed before and we need to retry.
NOTE: the projectors skip old events by looking to their subscriptions.

### CODE REVIEW

**Review by nelsonkopliku — APPROVED** [review]
Great

File: web/services/collector.go [hunk 1]
```diff
@@ -19,11 +22,36 @@ func NewCollectorService(db *gorm.DB, projectorsChannel chan *datapipeline.DataC
...
+			}
+			c.projectorsChannel <- collectedData
+			return nil
+		}
+
+		if err != nil {
+			return err
+		}
+
+		log.Debugf("Event already exists. Agent ID: %s, DiscoveryType: %s ", collectedData.AgentID, collectedData.DiscoveryType)
```
Comment (dottorblaster) [inline web/services/collector.go:52]: `log.Debugf` or `log.Warnf`?

**Review by dottorblaster — APPROVED** [review]
LGTM

File: web/services/collector.go (hunk 1 above)
Comment (fabriziosestito) [reply web/services/collector.go:52]: Debug i would say, this happens very often and we can't consider it a warning

**Review by nelsonkopliku — APPROVED** [review]

File: web/services/collector.go (hunk 1 above)
Comment (dottorblaster) [reply web/services/collector.go:52]: Ok!

**Review by rtorrero — APPROVED** [review]
LGTM

File: web/services/collector.go [hunk 2]
```diff
@@ -19,11 +22,36 @@ func NewCollectorService(db *gorm.DB, projectorsChannel chan *datapipeline.DataC
...
+		c.projectorsChannel <- &event
```
Comment (stefanotorresi) [inline web/services/collector.go:53]: I'm not quite clear, why are we projecting the last event?

Comment (fabriziosestito) [reply web/services/collector.go:53]: We still do not have a mechanism of retrial or queue persistence in the projector system, so if the event wasn't projected for some reason we need to be sure that the projectors will receive the event and try again.
The projectors keep the state of the subscription so if the event was already projected they will just skip it.

**Pull request merged**